src/ort_genai_c.h

```diff
@@ -170,38 +215,52 @@ OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsTryGraphCaptureWithMaxBatch
  * \param[in] generator_params The generator params to set the input ids on.
  * \param[in] sequences The input id sequences.
  * \return OgaResult containing the error message if the setting of the input id sequences failed.
  */
-OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsSetInputSequences(OgaGeneratorParams* generator_params, const OgaSequences* sequences);
+OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsSetInputSequences(OgaGeneratorParams* generator_params,
+                                                                       const OgaSequences* sequences);
 
-OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsSetInputs(OgaGeneratorParams* generator_params, const OgaNamedTensors* named_tensors);
+OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsSetInputs(OgaGeneratorParams* generator_params,
+                                                               const OgaNamedTensors* named_tensors);
 
 /*
- * \brief For additional model inputs that genai does not handle, this lets the user set their values. For example LoRA models handle
- * fine tuning through model inputs. This lets the user supply the fine tuning inputs, while genai handles the standard inputs.
- * \param[in] generator_params The generator params to set the input on
- * \param[in] name Name of the model input (this must match the model's input name)
- * \param[in] tensor The OgaTensor of the input data
+ * \brief For additional model inputs that genai does not handle, this lets the user set their values. For example LoRA
+ * models handle fine tuning through model inputs. This lets the user supply the fine tuning inputs, while genai handles
+ * the standard inputs. \param[in] generator_params The generator params to set the input on \param[in] name Name of the
+ * model input (this must match the model's input name) \param[in] tensor The OgaTensor of the input data
  */
-OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsSetModelInput(OgaGeneratorParams* generator_params, const char* name, OgaTensor* tensor);
+OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsSetModelInput(OgaGeneratorParams* generator_params,
+                                                                   const char* name, OgaTensor* tensor);
 
 OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsSetWhisperInputFeatures(OgaGeneratorParams*, OgaTensor* tensor);
 
+/// <summary>
+/// Set active adapter names for the give params instance
+/// </summary>
+/// <param name="">a valid params ptr</param>
+/// <param name="adapters">an array of valid adapter names (UTF-8 encoded C strings)</param>
+/// <param name="count">size of the names array</param>
+/// <returns>Returns an error if a non-existent adapter name is submitted</returns>
+OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsSetActiveAdapters(OgaGeneratorParams* params,
+                                                                       const char* const* adapters, size_t count);
+
```

Comment:
Could we add a mechanism to query currently active adapters?
Also, should we add a mechanism to deactivate (1 or more) currently active adapters?

Reply:
For the first one, see my response to a similar comment.

For the second one, GeneratorParams along with the generator is an ephemeral structure that represents a configuration for a given generator session.

Once generator is created, it creates its session state where the appropriate lora inputs and other extra inputs are recorded. You cannot change it after the fact.

To run with another set of adapters (or none) one needs to create a different Generator.

You can reset the set of adapters within Generator Params by passing either a different set of adapter names OR pass an empty set.